libnetwork/drivers/overlay/peerdb.go

```diff
@@ -63,7 +63,7 @@ func (p *peerEntryDB) UnMarshalDB() peerEntry {
 
 type peerMap struct {
 	// set of peerEntry, note they have to be objects and not pointers to maintain the proper equality checks
-	mp setmatrix.SetMatrix
+	mp *setmatrix.SetMatrix
```

Comment:
Just to be sure: the above comment relates to the objects stored by `SetMatrix`, not the `mp` variable itself?

I got confused a bit by this one, maybe we could move this comment to the godoc of `SetMatrix` or `SetMatrix.Insert`?

Reply:
Yes, the comment relates to the `peerEntry` values inserted into the `SetMarix`. Moving the comment to the godoc of `SetMatrix` wouldn't be appropriate as it is not universally applicable advice for a general purpose container. Reference equality could be appropriate in different use-cases. The ideal solution would be to make `SetMatrix` a generic type so that "insert `peerEntry` not `*peerEntry`" can be expressed in a way which the compiler and IDEs understand. I'll do that in a follow-up PR.